ingest/author_de.py

```diff
@@ -47,32 +35,45 @@ def sort_comparison(groups):
         return sorted(groups)
 
 
-def convert_seurat_findallmarkers_to_wide(data):
-    """Convert from Seurat FindAllMarkers() format to SCP DE wide format
-
-    TODO (SCP-5296): Finish handling
+def canonicalize_name_and_order(data, header_refmap):
+    """Ensure dataframe has column names and ordering as expected by DE UI
     """
-    data = data.rename(columns={"cluster": "group", "gene": "genes"})
+
+    is_one_vs_rest_only = header_refmap["comparison_group"] == "None"
+    if is_one_vs_rest_only:
+        data = data.assign(comparison_group="rest")
+
+    # Update column names in DF to expected header names
+    rename_map = {}
+    for ref_header in header_refmap:
+        if ref_header in ["size", "significance"]:
+            continue
+        actual_header = header_refmap[ref_header]
+        rename_map[actual_header] = ref_header
+    data = data.rename(columns=rename_map)
+
     data = data.astype({"group": "string"})
-    data = data.assign(comparison_group="rest")
-    expected_header_order = [
-        "genes", "group", "comparison_group", "avg_log2FC", "p_val_adj", "p_val", "pct.1", "pct.2"
-    ]
-    data = data[expected_header_order]
-    wide_df = convert_long_to_wide(data)
-    return wide_df
+
+    # Update headers to expected order
+    unsorted_headers = list(data.columns)
+    size = header_refmap["size"]
+    significance = header_refmap["significance"]
+    sorted_headers = sort_headers(unsorted_headers, size, significance)
+    canonical_long_df = data[sorted_headers]
+
+    return canonical_long_df
 
 
 def convert_long_to_wide(data):
-    """Convert from long format to SCP DE wide format
+    """Convert from canonical DE long format to SCP DE wide format
 
     (Long format is typical uploaded, but this module internally uses wide.)
     """
     metrics = list(data.columns[3:])
     data["combined"] = data["group"] + "--" + data["comparison_group"]
     frames = []
     # E.g.
-    # gene  group   comparison_group    log2foldchange  pvals_adj   qvals   mean    cat dog
+    # genes  group   comparison_group    log2foldchange  pvals_adj   qvals   mean    cat dog
```

Comment:
Should the column headers either all be singular or all be plural?

singular: gene group comparison_group log2foldchange pval_adj qval mean cat dog

plural: genes groups comparison_groups log2foldchanges pvals_adj qvals means cats dogs

Most internet sources seem to advocate for singular unless the column in question allows multiple values (ie an array). I found enough of the reasons listed in [this StackOverflow thread](https://stackoverflow.com/questions/338156/table-naming-dilemma-singular-vs-plural-names) to be intuitive that my personal preference would be for singular but there are also reasonable arguments for plural. I suggest choosing one form and being consistent.

Reply:
Astute point!  That SO thread is insightful.  Per chat, it seems noteworthy that Seurat uses singular, whereas Scanpy uses plural.  I've gone ahead and updated "genes" to "gene" in the canonical output names, and column header value.  

This will make the transformed output in that column header more consistent, and also make our corresponding internal CLI arguments more consistent.